source/retryable-reads/retryable-reads.rst

```diff
@@ -163,6 +163,7 @@ Drivers SHOULD support retryability for the following operations:
 - Any driver that provides generic command runners for read commands (with logic
   to inherit a client-level read concerns) SHOULD implement retryability for the
   read-only command runner.
+- Handshake failures due to a network error.
```

Comment:
Changed the wording but its kinda light still. I'm not fully grokking how you determine its a handshake failure at this layer. Would it be correct to say any error that checkOut throws that isn't related to pool errors or waitQueueTimeouts is retryable? and implicitly related to handshakes in that case?

Currently the pseudo code catches a "NetworkError" but am I correct in assuming a ShutdownInProgress error doesn't reset the connection? So we need to add logic/spec text to pass errors from `pool.checkOut()` through our exising retryable rules, since that should catch something like shutdownInProgress

Reply:
I'll put up a test to include the ShutDownInProgress example we discussed, I think it would fit here.